crates/type-system/scripts/build-wasm.ts

```diff
@@ -0,0 +1,80 @@
+import fs from "fs-extra";
+import * as child_process from "node:child_process";
+import * as path from "node:path";
+import { fileURLToPath } from "node:url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// The PathObject definition in Node's path package
+type PathObject = {
+  dir: string;
+  root: string;
+  base: string;
+  name: string;
+  ext: string;
+};
+
+const runWasmPack = (target: string) => {
+  console.log(`Running wasm-pack targeting ${target}`);
+  const { error } = child_process.spawnSync("wasm-pack", [
+    "build",
+    "--target",
+    target,
+    "--out-name",
+    "index",
+    "--scope",
+    "blockprotocol",
+  ]);
+  if (error) {
+    throw error;
+  }
+};
+
+const moveSrcFiles = (packagePath: PathObject, packageName: string) => {
+  const destinationPath = path.resolve(
+    __dirname,
+    "..",
+    "..",
+    "..",
```

Comment:
Potential future improvement: might be worth avoiding `__dirname` in ESM modules. It's a built-in constant, which may look confusing (are we in CommonJS? 🤔)

We rarely  need script dir name, so can go for something like:

```ts
export const packageRootDirPath = path.resolve(
  path.dirname(fileURLToPath(import.meta.url)),
  "..",
);
```

(copied from some other file)